Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -0,0 +1,324 @@
+
+            // The full source path asset exists on disk, so use that as
+            // the pre-selected file when the user opens the file picker dialog
+            if (fullPathIsValid)
+            {
+                preselectedFilePath = fullPath.c_str();
+            }
+            // GetFullSourcePathFromRelativeProductPath failed so the file doesn't exist on disk yet
+            // So we need to find it by searching in the asset safe folders
+            else
+            {
+                bool assetSafeFoldersRetrieved = false;
+                AZStd::vector<AZStd::string> assetSafeFolders;
+                AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                    assetSafeFoldersRetrieved, &AzToolsFramework::AssetSystemRequestBus::Events::GetAssetSafeFolders, assetSafeFolders);
+
+                if (!assetSafeFoldersRetrieved)
+                {
+                    AZ_Error("PropertyFilePathCtrl", false, "Could not acquire a list of asset safe folders from the database.");
+                    return QString();
+                }
+
+                // Find an asset safe folder that already has existing sub-path that
+                // would satisfy the relative path for the file
+                for (const AZStd::string& assetSafeFolder : assetSafeFolders)
+                {
+                    AZ::IO::Path assetSafeFolderPath(assetSafeFolder);
+                    AZ::IO::Path relativeFolderPath = assetSafeFolderPath;
+
+                    if (m_currentFilePath.HasParentPath())
+                    {
+                        relativeFolderPath /= m_currentFilePath.ParentPath();
+                    }
+
+                    QDir currentFilePathDirectory(relativeFolderPath.c_str());
+                    if (currentFilePathDirectory.exists())
+                    {
+                        auto absoluteFilePath = (assetSafeFolderPath / m_currentFilePath);
+                        preselectedFilePath = absoluteFilePath.c_str();
+                        break;
+                    }
+                }
+            }
+        }
+
+        return preselectedFilePath;
+    }
+
+    void PropertyFilePathCtrl::HandleNewFilePathSelected(const QString& newFilePath)
+    {
+        // If the newFilePath is empty, then the dialog was cancelled,
+        // so don't process any further.
+        if (newFilePath.isEmpty())
+        {
+            return;
+        }
+
+        // Generate a relative path given the absolute path that was picked.
+        // If the absolute path wasn't inside an asset safe folder, then
+        // the call to GenerateRelativeSourcePath will fail.
+        AZStd::string absolutePath(newFilePath.toUtf8().constData());
+        AZStd::string relativePath, rootFilePath;
+        bool relativePathIsValid;
+        AssetSystemRequestBus::BroadcastResult(
+            relativePathIsValid, &AssetSystemRequestBus::Events::GenerateRelativeSourcePath, absolutePath, relativePath, rootFilePath);
```

Comment:
The `relativePathIsValid` must be initialized.
AssetSystemRequestBus might not be connected and this would result in `relativePathIsValid` never having a valid value

Reply:
Good call, fixed